core/src/main/java/org/apache/calcite/sql/fun/SqlLibraryOperators.java

```diff
@@ -878,6 +880,28 @@ private static RelDataType arrayReturnType(SqlOperatorBinding opBinding) {
           SqlLibraryOperators::arrayReturnType,
           OperandTypes.SAME_VARIADIC);
 
+  @SuppressWarnings("argument.type.incompatible")
+  private static RelDataType arrayAppendPrependReturnType(SqlOperatorBinding opBinding) {
```

Comment:
Is there no more suitable class to save it?

`SqlLibraryOperators` only should  defines functions and operators.

Reply:
@JiajunBernoulli, I think it's OK. There are already a few methods like this that help define the return type (and are used via  `::` method references to become lambdas)

Reply:
Thanks for your explanation.